crates/metadata-server/src/raft/network/networking.rs

```diff
@@ -103,23 +103,28 @@ where
                 .try_send(network_message)
                 .map_err(|_err| TrySendError::Send(message))?;
         } else if let Some(address) = self.addresses.get(&target) {
-            if let Some(task_handle) = self.connection_attempts.remove(&target) {
-                if !task_handle.is_finished() {
+            if let Entry::Occupied(occupied) = self.connection_attempts.entry(target) {
+                if !occupied.get().is_finished() {
                     return Err(TrySendError::Connecting(message));
                 } else {
-                    match task_handle.now_or_never().expect("should be finished") {
-                        Ok(result) => match result {
+                    match occupied.remove().now_or_never() {
```

Comment:
you can also use `unconstrained()` but it's always safer to not panic anyway.

Reply:
Good point. I'll remember it for the future :-)